asyncpg/protocol/record/recordobj.c

```diff
@@ -254,6 +254,55 @@ record_item(ApgRecordObject *o, Py_ssize_t i)
 }
 
 
+/* Lookup a record value by its name.  Return 0 on success, -2 if the
+ * value was not found (with KeyError set), and -1 on all other errors.
+ */
+static int
+record_item_by_name(ApgRecordObject *o, PyObject *item, PyObject **result)
+{
+    PyObject *mapped;
+    PyObject *val;
+    Py_ssize_t i;
+
+    mapped = PyObject_GetItem(o->desc->mapping, item);
+    if (mapped == NULL) {
+        goto noitem;
+    }
+
+    if (!PyIndex_Check(mapped)) {
+        Py_DECREF(mapped);
+        goto error;
+    }
+
+    i = PyNumber_AsSsize_t(mapped, PyExc_IndexError);
+    Py_DECREF(mapped);
+
+    if (i < 0) {
+        if (PyErr_Occurred())
+            PyErr_Clear();
+        goto error;
+    }
+
+    val = record_item(o, i);
+    if (val == NULL) {
+        PyErr_Clear();
+        goto error;
+    }
+
+    *result = val;
+
+    return 0;
+
+noitem:
+    _PyErr_SetKeyError(item);
+    return -2;
```

Comment:
Let's use an enum type, please. It's hard to read the code with checks for -1 or -2.

Reply:
Done